File: src/exemplos/04_Subalgoritmos/03-escopo.py
#  -*- coding: utf-8 -*-
##   @package: 03-escopo.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo de escopos de variáveis locais.


def troca(x, y):
    '''Troca os valores de x e y no escopo local.'''
    x, y = y, x
    print('troca: (x,y) = (%d,%d)' % (x, y))


x = 1
y = 2

print('(x,y) = (%d,%d)' % (x, y))
print('Trocando...', end=' ')
troca(x, y)
print('Trocou!')
print('(x,y) = (%d,%d)' % (x, y))

File: src/exemplos/04_Subalgoritmos/03-escopo/03-escopo.c
/**      @file: 03-escopo.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo de escopos de variáveis locais. */

#include <stdio.h>

/* Troca os valores de x e y no escopo local. */
void troca(int x, int y) {
  int aux = x;
  x = y;
  y = aux;

  printf("troca: (x,y) = (%d,%d)\n", x, y);
}

int main() {
    int x = 1;
    int y = 2;

    printf("(x,y) = (%d,%d)\n", x, y);
    printf("Trocando... ");
    troca(x, y);
    printf("Trocou!\n");
    printf("(x,y) = (%d,%d)\n", x, y);

  return 0;
}